### PR TITLE
186658870 post exit assessment r20

### PR DESCRIPTION
**Pull request by ttoomey**
## Description

[Support Post-Exit assessment for R20 Aftercare
](https://www.pivotaltracker.com/story/show/186658870)

[//]: # 'remove if not applicable'
[Depends on hmis-warehouse PR:
](https://github.com/greenriver/hmis-warehouse/pull/3901)

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 1]
```diff
@@ -92,9 +97,14 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
...
 
-  if (isEmpty(items)) return null;
+  if (items.length === 0)
+    return (
+      <LoadingButton loading={loading} disabled={!loading}>
+        Assessments
+      </LoadingButton>
+    );
 
   return <CommonMenuButton title='Assessments' items={items} />;
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]:
```suggestion
  return <CommonMenuButton title='New Assessment' items={items} />;
```

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 2]
```diff
@@ -55,35 +70,25 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
...
-        to: getPath(AssessmentRole.Exit),
-        title: 'HUD Exit Assessment',
+    if (!data?.client) return [];
+
+    const hudRoleTitles: Partial<Record<AssessmentRole, string>> = {
+      INTAKE: 'HUD Intake Assessment',
+      UPDATE: 'New HUD Update Assessment',
+      EXIT: 'New HUD Exit Assessment',
+      ANNUAL: 'New HUD Annual Assessment',
+      POST_EXIT: 'New HUD Post-Exit Assessment',
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]:
```suggestion
      INTAKE: 'Intake Assessment',
      UPDATE: 'Update Assessment',
      EXIT: 'Exit Assessment',
      ANNUAL: 'Annual Assessment',
      POST_EXIT: 'Post-Exit Assessment',
```

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 3]
```diff
@@ -55,35 +70,25 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
...
+    };
+
+    data.client.assessmentEligibilities.forEach(({ id, role, title }) => {
+      const items =
+        role === 'INTAKE' || role === 'EXIT' ? topItems : bottomItems;
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: remove this, change them to be all in one section with no divider.


this was an attempt to separate out intake/exit as special types of assessments that can't have multiples. but it feels cluttered and unclear, so I think we should just simplify.

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 4]
```diff
@@ -92,9 +97,14 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
         : []),
       ...bottomItems,
     ];
-  }, [enrollment, getPath]);
+  }, [data, getPath]);
 
-  if (isEmpty(items)) return null;
+  if (items.length === 0)
+    return (
+      <LoadingButton loading={loading} disabled={!loading}>
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: Nit: it would be nice if this was a skeleton while loading. Imo the loading spinner on the button evokes the idea that the button is performing some action (meaning I clicked it) since that's how we display loading submissions elsewhere.

Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: Nit: it would be nice if there was a tooltip explaining `Cannot to perform new assessments for exited enrollment.` if it is disabled. I we use the ButtonTooltipContainer for that

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 5]
```diff
@@ -41,6 +45,17 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
   const enrollmentId = enrollment.id;
   const clientId = enrollment.client.id;
 
+  const [getEligability, { data, error, loading }] =
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: spellcheck: `getEligibility`

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx [hunk 6]
```diff
@@ -41,6 +45,17 @@ const NewAssessmentMenu: React.FC<Props> = ({ enrollment }) => {
...
+    useGetClientAssessmentEligibilitiesLazyQuery({
+      fetchPolicy: 'cache-and-network',
+    });
+
+  useEffect(() => {
+    getEligability({ variables: { clientId, enrollmentId } });
+  }, [enrollment.client.lockVersion, clientId, enrollmentId, getEligability]);
```
Comment (gigxz) [inline src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: oh, why did you add the lock version like this?

**Review by gigxz — COMMENTED** [review]
I have some scope-creeping style suggestions, otherwise looks good!

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx (hunk 6 above)
Comment (ttoomey) [reply src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: This was a quick way to ensure the query is run again after an assessment is added. I'll revisit this and see if it's still necessary

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx (hunk 4 above)
Comment (ttoomey) [reply src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: oh good point. Yeah skeleton would be better, I'll update it

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx (hunk 2 above)
Comment (ttoomey) [reply src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: These seem to the the titles we get from the forms on the backend. Looks like we don't need to override the titles at all in this case.

File: src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx (hunk 6 above)
Comment (ttoomey) [reply src/modules/enrollment/components/EnrollmentAssessmentActionButtons.tsx]: looks like it's not needed with a cache-and-network fetch policy

**Review by gigxz — APPROVED** [review]
💯
